From 7a47de6cd61e311fe67fc61b7d3ec94a04c24142 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Thu, 30 May 2019 15:12:45 +0800
Subject: [PATCH] Revert "drm/gem: drm_gem_dumb_map_offset(): reject dma-buf"

This reverts commit 90378e589192 ("drm/gem: drm_gem_dumb_map_offset(): reject dma-buf").

There are 2 drm devices in i.mx8mq, one is DCSS and the other is
GPU (GC7000 Lite). When starting weston service in userspace, it will
call libgbm to let GC7000 to allocate dma_buf as its buffer, and then
GC7000 will pass a fd which associated to this dma_buf to DCSS.

Now DCSS use the fd passed by GC7000 to find the dma_buf and import it
into its own context. This is implementation of the buffer sharing
between graphic cards.

In upstream commit 90378e589192 ("drm/gem: drm_gem_dumb_map_offset(): reject dma-buf"),
says that it is not allowed that DCSS use the imported dma_buf as a dumb map
buffer.

But the libgbm which provided by NXP uses this unsuggested method and because
it's binary format, we can do nothing but only change kernel source as workaround.

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/gpu/drm/drm_gem.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/drivers/gpu/drm/drm_gem.c b/drivers/gpu/drm/drm_gem.c
index 4a16d7b26c89..305e8d451945 100644
--- a/drivers/gpu/drm/drm_gem.c
+++ b/drivers/gpu/drm/drm_gem.c
@@ -326,12 +326,6 @@ int drm_gem_dumb_map_offset(struct drm_file *file, struct drm_device *dev,
 	if (!obj)
 		return -ENOENT;
 
-	/* Don't allow imported objects to be mapped */
-	if (obj->import_attach) {
-		ret = -EINVAL;
-		goto out;
-	}
-
 	ret = drm_gem_create_mmap_offset(obj);
 	if (ret)
 		goto out;
-- 
2.17.1

